clients/src/main/java/org/apache/kafka/common/requests/VoteResponse.java

```diff
@@ -84,15 +84,20 @@ public static VoteResponseData singletonResponse(Errors topLevelError,
     @Override
     public Map<Errors, Integer> errorCounts() {
         Map<Errors, Integer> errors = new HashMap<>();
-        Errors topLevelError = Errors.forCode(data.errorCode());
-        if (topLevelError != Errors.NONE) {
-            errors.put(topLevelError, 1);
-        }
+
+        int numPartitions = 0;
         for (VoteResponseData.VoteTopicResponse topicResponse : data.topics()) {
             for (VoteResponseData.VotePartitionResponse partitionResponse : topicResponse.partitions()) {
                 errors.compute(Errors.forCode(partitionResponse.errorCode()),
                     (error, count) -> count == null ? 1 : count + 1);
             }
+            numPartitions += topicResponse.partitions().size();
+        }
+
+        // The top-level error applies to all partitions.
+        Errors topLevelError = Errors.forCode(data.errorCode());
+        if (topLevelError != Errors.NONE) {
+            errors.put(topLevelError, numPartitions);
```

Comment:
Hmm.. What I was trying to get to is that each partition in the request would be reflected exactly once in this error map. But I guess that cannot be achieved with the change to `getErrorResponse` since we do not return all partitions in the response. Our options are either to revert the logic in `getErrorResponse` or just accept that the interpretation of these errors must take into account whether it is a top-level error code or not. I'm leaning toward the latter, in which case, maybe it is better to restore the original logic which added a count of 1 for the top-level error.

Reply:
Make sense, will revert.